source/frontend/src/features/properties/map/compensation/update/UpdateCompensationRequisitionForm.tsx

```diff
@@ -1,39 +1,63 @@
-import { FastDatePicker, Select, SelectOption, TextArea } from 'components/common/form';
+import {
+  Check,
+  FastCurrencyInput,
+  FastDatePicker,
+  Input,
+  Select,
+  SelectOption,
+  TextArea,
+} from 'components/common/form';
 import { UnsavedChangesPrompt } from 'components/common/form/UnsavedChangesPrompt';
 import LoadingBackdrop from 'components/maps/leaflet/LoadingBackdrop/LoadingBackdrop';
 import { Section } from 'features/mapSideBar/tabs/Section';
 import { SectionField } from 'features/mapSideBar/tabs/SectionField';
 import { Formik, FormikProps } from 'formik';
 import { getCancelModalProps, useModalContext } from 'hooks/useModalContext';
 import { Api_AcquisitionFile } from 'models/api/AcquisitionFile';
-import { Api_Compensation } from 'models/api/Compensation';
+import { Api_CompensationRequisition } from 'models/api/CompensationRequisition';
 import moment from 'moment';
+import { useEffect, useRef, useState } from 'react';
 import { Prompt } from 'react-router-dom';
 import styled from 'styled-components';
+import { withNameSpace } from 'utils/formUtils';
 
 import SidebarFooter from '../../shared/SidebarFooter';
 import { CompensationRequisitionYupSchema } from '../CompensationRequisitionYupSchema';
 import { CompensationRequisitionFormModel } from '../models';
+import FinancialActivitiesSubForm from './financials/FinancialActivitiesSubForm';
 
 export interface CompensationRequisitionFormProps {
   isLoading: boolean;
-  formikRef: React.Ref<FormikProps<CompensationRequisitionFormModel>>;
   acquisitionFile: Api_AcquisitionFile;
   initialValues: CompensationRequisitionFormModel;
-  onSave: (compensation: CompensationRequisitionFormModel) => Promise<Api_Compensation | undefined>;
+  gstConstant: number;
+  financialActivityOptions: SelectOption[];
+  chartOfAccountsOptions: SelectOption[];
+  responsiblityCentreOptions: SelectOption[];
+  yearlyFinancialOptions: SelectOption[];
+  onSave: (
+    compensation: CompensationRequisitionFormModel,
+  ) => Promise<Api_CompensationRequisition | undefined>;
   onCancel: () => void;
 }
 
 const UpdateCompensationRequisitionForm: React.FC<CompensationRequisitionFormProps> = ({
   isLoading,
-  formikRef,
   acquisitionFile,
   initialValues,
+  gstConstant,
+  financialActivityOptions,
+  chartOfAccountsOptions,
+  responsiblityCentreOptions,
+  yearlyFinancialOptions,
   onSave,
   onCancel,
 }) => {
+  const fiscalYearOptions = generateFiscalYearOptions();
   const { setModalContent, setDisplayModal } = useModalContext();
-  let fiscalYearOptions = generateFiscalYearOptions();
+  const formikRef = useRef<FormikProps<CompensationRequisitionFormModel>>(null);
+  const dummyOptions: SelectOption[] = [];
+  const [activitiesUpdated, setActivitiesUpdated] = useState<boolean>(false);
```

Comment:
lets stick with our patterns and not track state within non-container components.

Reply:
is the same patter as above? need this to keep track of activity updates.